hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/JpaRolloutGroupManagement.java

```diff
@@ -268,6 +273,25 @@ public Page<TargetWithActionStatus> findAllTargetsOfRolloutGroupWithActionStatus
         return new PageImpl<>(targetWithActionStatus, pageRequest, totalCount);
     }
 
+    private List<Order> getOrderBy(final Pageable pageRequest, final CriteriaBuilder cb,
+            final Join<RolloutTargetGroup, JpaTarget> targetJoin,
+            final ListJoin<RolloutTargetGroup, JpaAction> actionJoin) {
+        final List<Order> orders = new ArrayList<>();
+
+        pageRequest.getSort().get().forEach(order -> {
+            final String property = order.getProperty();
+            // we consider status as property from JpaAction ...
+            if ("status".equals(property)) {
+                orders.addAll(QueryUtils.toOrders(Sort.by(order.getDirection(), property), actionJoin, cb));
+            }
+            // ... and every other property from JpaTarget
+            else {
+                orders.addAll(QueryUtils.toOrders(Sort.by(order.getDirection(), property), targetJoin, cb));
+            }
+        });
+        return orders;
```

Comment:
```suggestion
        return pageRequest.getSort().get().flatMap(order -> {
            final String property = order.getProperty();
            // we consider status as property of JpaAction ...
            final List<Order> orders;
            if ("status".equals(property)) {
                orders = QueryUtils.toOrders(Sort.by(order.getDirection(), property), actionJoin, cb);
            }
            // ... and every other property of JpaTarget
            else {
                orders = QueryUtils.toOrders(Sort.by(order.getDirection(), property), targetJoin, cb);
            }
            return orders.stream();
        }).collect(Collectors.toList());
```

Reply:
Explanation regarding suggestion above: If you want to avoid the explicit list creation, you can use flatMap + collect.